pkg/skaffold/cluster/minikube.go

```diff
@@ -24,21 +24,36 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"sync"
 
+	"github.com/blang/semver"
 	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/util/homedir"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/version"
+)
+
+const (
+	minikubeVersionWithUserFlag = "1.18.0"
```

Comment:
Since this is our string, we can just use `MustParse()` here
```suggestion
	minikubeVersionWithUserFlag = semver.MustParse("1.18.0")
```

and then turn `supportsUserFlag()` into a predicate:
```
func supportsUserFlag(ver semver.Version) bool {
	return ver.GE(minikubeVersionWithUserFlag)
}
```

with the cascading changes.